x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobDataDeleter.java

```diff
@@ -160,7 +172,13 @@ public void deleteAnnotations(
     ) {
         BoolQueryBuilder boolQuery = QueryBuilders.boolQuery()
             .filter(QueryBuilders.termQuery(Job.ID.getPreferredName(), jobId))
-            .filter(QueryBuilders.termQuery(Annotation.CREATE_USERNAME.getPreferredName(), XPackUser.NAME));
+            .filter(
+                (deleteUserAnnotations)
+                    ? QueryBuilders.boolQuery()
+                        .should(QueryBuilders.termQuery(Annotation.CREATE_USERNAME.getPreferredName(), XPackUser.NAME))
+                        .should(QueryBuilders.termQuery(Annotation.CREATE_USERNAME.getPreferredName(), ElasticUser.NAME))
+                    : QueryBuilders.termQuery(Annotation.CREATE_USERNAME.getPreferredName(), XPackUser.NAME)
```

Comment:
If `deleteUserAnnotations` is true then it just shouldn't add a filter on the username.  That will make it delete everything.